### ISSUE TITLE
False positive of S4165

### ISSUE THREAD

**Issue by zloirock** [SonarSource/SonarJS#4767]
I'm playing with `SonarJS` with `eslint-plugin-sonar` and see this false positive of S4165:
```js
var dedentStringsArray = function (template) {
  // ...
  for (var i = 0; i < length; i++) {
    // ...
    for (var j = 2; j < lines.length; j += 2) { // <- Review this redundant assignment: "j" already holds the assigned value along all execution paths  sonar/no-redundant-assignments
      // ...
    }
  }
  // ...
};
```

`j` is not assigned in any other place.

**Comment by ilia-kebets-sonarsource** [comment]
Hey @zloirock,
We are moving away from GitHub issues at Sonar and suggest our users go through our community website to submit issues  of all kinds, including false positive, as you can see [when clicking on new issue](https://github.com/SonarSource/SonarJS/issues/new/choose).

Could you please recreate your issue there? It should be triaged and assigned to our team quickly.

Also, since we changed the behaviour of the "new issue" button to [this](https://github.com/SonarSource/SonarJS/issues/new/choose), I was wondering how did you manage to create an issue?

**Comment by zloirock** [comment]
> Could you please recreate your issue there?

Ok.

> how did you manage to create an issue?

https://github.com/SonarSource/SonarJS/issues/new

**Comment by zloirock** [comment]
https://community.sonarsource.com/t/false-positive-of-s4165/121062

**Comment by ilia-kebets-sonarsource** [comment]
closing, please follow on community
